pkg/processor/runtime/rpc/abstract.go

```diff
@@ -143,13 +153,65 @@ func (r *AbstractRuntime) ProcessEvent(event nuclio.Event, functionLogger logger
 		r.functionLogger = nil
 		return nil, errors.New(msg)
 	}
-
+	if processingResult.err != nil {
+		return nil, processingResult.err
+	}
+	// this is a single event processing flow, so we only take the first item from the result
 	return nuclio.Response{
-		Body:        result.DecodedBody,
-		ContentType: result.ContentType,
-		Headers:     result.Headers,
-		StatusCode:  result.StatusCode,
-	}, result.err
+		Body:        processingResult.results[0].DecodedBody,
+		ContentType: processingResult.results[0].ContentType,
+		Headers:     processingResult.results[0].Headers,
+		StatusCode:  processingResult.results[0].StatusCode,
+	}, processingResult.results[0].err
+}
+
+// ProcessBatch processes a batch of events
+func (r *AbstractRuntime) ProcessBatch(batch []nuclio.Event, functionLogger logger.Logger) ([]*runtime.ResponseWithErrors, error) {
+	if currentStatus := r.GetStatus(); currentStatus != status.Ready {
+		return nil, errors.Errorf("Processor not ready (current status: %s)", currentStatus)
+	}
+
+	r.functionLogger = functionLogger
+
+	// We don't use defer to reset r.functionLogger since it decreases performance
+	if err := r.eventEncoder.Encode(batch); err != nil {
+		r.functionLogger = nil
+		return nil, errors.Wrapf(err, "Can't encode batch: %+v", batch)
+	}
+
+	processingResults, ok := <-r.resultChan
+	r.functionLogger = nil
+	if !ok {
+		msg := "Client disconnected"
+		r.Logger.Error(msg)
+		r.SetStatus(status.Error)
+		r.functionLogger = nil
+		return nil, errors.New(msg)
+	}
+	responsesWithErrors := make([]*runtime.ResponseWithErrors, len(processingResults.results))
+
+	// if processingResults.err is not nil, it means that whole batch processing was failed
+	if processingResults.err != nil {
+		return nil, processingResults.err
+	}
+
+	for index, processingResult := range processingResults.results {
+		if processingResult.EventId == "" {
+			functionLogger.WarnWith("Received response with empty event_id, response won't be returned")
```

Comment:
if it won't be returned, do you still need to put it in the map below or can you `continue`?